implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/v2/DynatraceExporterV2Test.java

```diff
@@ -0,0 +1,380 @@
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).startsWith("my.summary,dt.metrics.source=micrometer gauge");
+        assertThat(actual.get(0)).contains("max=").contains("min=").contains("sum=").contains("count=");
+    }
+
+    @Test
+    void toGaugeLine() {
+        meterRegistry.gauge("my.gauge", 1.23);
+        Gauge myGauge = meterRegistry.find("my.gauge").gauge();
+
+        List<String> actual = exporter.toGaugeLine(myGauge).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).startsWith("my.gauge,dt.metrics.source=micrometer gauge,1.23 ");
+        String expectedDummy = "my.gauge,dt.metrics.source=micrometer gauge,1.23 1617714022879";
+        assertThat(actual.get(0)).hasSize(expectedDummy.length());
+    }
+
+
+    @Test
+    void toMeterLine() {
+        meterRegistry.gauge("my.meter", 1.23);
+        Gauge gauge = meterRegistry.find("my.meter").gauge();
+        assertNotNull(gauge);
+
+        List<String> actual = exporter.toMeterLine(gauge).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).startsWith("my.meter,dt.metrics.source=micrometer gauge,1.23");
+        String expectedDummy = "my.meter,dt.metrics.source=micrometer gauge,1.23 1617714022879";
+        assertThat(actual.get(0)).hasSize(expectedDummy.length());
+    }
+
+    @Test
+    void toCounterLine() {
+        meterRegistry.counter("my.counter");
+        Counter counter = meterRegistry.find("my.counter").counter();
+        assertNotNull(counter);
+        counter.increment();
+        counter.increment();
+        counter.increment();
+
+        // wait for the next export interval
+        try {
+            Thread.sleep(300);
+        } catch (InterruptedException ignored) {
+        }
+
+        String expected = "my.counter,dt.metrics.source=micrometer count,delta=3.0 ";
+        List<String> actual = exporter.toCounterLine(counter).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        String actualLine = actual.get(0);
+        String expectedDummy = "my.counter,dt.metrics.source=micrometer count,delta=3.0 1617714022879";
+        assertThat(actualLine).hasSize(expectedDummy.length());
+        assertThat(actualLine).startsWith(expected);
+    }
+
+    @Test
+    void toTimerLine() {
+        meterRegistry.timer("my.timer");
+        Timer timer = meterRegistry.find("my.timer").timer();
+        assertNotNull(timer);
+
+        timer.record(Duration.ofMillis(60));
+
+
+        try {
+            Thread.sleep(300);
+        } catch (InterruptedException ignored) {
+        }
+
+        List<String> actual = exporter.toTimerLine(timer).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).startsWith("my.timer,dt.metrics.source=micrometer gauge,min=60.0,max=60.0,sum=60.0,count=1 ");
```

Comment:
You could also record two or more data points so you could detect values being mixed up at this point, currently it's three times the same.